packages/network-controller/src/NetworkController.ts

```diff
@@ -350,44 +484,76 @@ export type NetworkControllerMessenger = RestrictedControllerMessenger<
 
 export type NetworkControllerOptions = {
   messenger: NetworkControllerMessenger;
-  trackMetaMetricsEvent: () => void;
   infuraProjectId: string;
   state?: Partial<NetworkState>;
 };
 
-export const defaultState: NetworkState = {
-  selectedNetworkClientId: NetworkType.mainnet,
-  networksMetadata: {},
-  networkConfigurations: {},
-};
+/**
+ * Constructs a value for the state property `networkConfigurationsByChainId`
+ * which will be used if it has not been provided to the constructor.
+ *
+ * @returns The default value for `networkConfigurationsByChainId`.
+ */
+function getDefaultNetworkConfigurationsByChainId(): Record<
+  Hex,
+  NetworkConfiguration
+> {
+  return Object.values(InfuraNetworkType).reduce(
+    (obj: Partial<Record<Hex, NetworkConfiguration>>, infuraNetworkType) => {
+      const chainId = ChainId[infuraNetworkType];
+      const rpcEndpointUrl = `https://${infuraNetworkType}.infura.io/v3/{infuraProjectId}`;
+
+      const networkConfiguration: NetworkConfiguration = {
+        chainId,
+        defaultRpcEndpointUrl: rpcEndpointUrl,
+        name: NetworkNickname[infuraNetworkType],
+        nativeTokenName: NetworksTicker[infuraNetworkType],
+        rpcEndpoints: [
+          {
+            name: `Infura ${NetworkNickname[infuraNetworkType] as string}`,
+            networkClientId: infuraNetworkType,
+            type: RpcEndpointType.Infura,
+            url: rpcEndpointUrl,
+          },
+        ],
+      };
 
-type MetaMetricsEventPayload = {
-  event: string;
-  category: string;
-  referrer?: { url: string };
-  actionId?: number;
-  environmentType?: string;
-  properties?: unknown;
-  sensitiveProperties?: unknown;
-  revenue?: number;
-  currency?: string;
-  value?: number;
-};
+      return { ...obj, [chainId]: networkConfiguration };
+    },
+    {},
+  ) as Record<Hex, NetworkConfiguration>;
+}
```

Comment:
It looks like we can get rid of the type asssertions here.

```suggestion
  return Object.values(InfuraNetworkType).reduce<Record<Hex, NetworkConfiguration>>(
    (obj, infuraNetworkType) => {
      const chainId = ChainId[infuraNetworkType];
      const rpcEndpointUrl = `https://${infuraNetworkType}.infura.io/v3/{infuraProjectId}`;

      const networkConfiguration: NetworkConfiguration = {
        chainId,
        defaultRpcEndpointUrl: rpcEndpointUrl,
        name: NetworkNickname[infuraNetworkType],
        nativeTokenName: NetworksTicker[infuraNetworkType],
        rpcEndpoints: [
          {
            name: `Infura ${NetworkNickname[infuraNetworkType]}`,
            networkClientId: infuraNetworkType,
            type: RpcEndpointType.Infura,
            url: rpcEndpointUrl,
          },
        ],
      };

      return { ...obj, [chainId]: networkConfiguration };
    },
    {},
  );
}
```

If you want to be strict about `obj`'s type during iteration, reverting line 502 from the above to the following also works:
```ts
    (obj: Partial<Record<Hex, NetworkConfiguration>>, infuraNetworkType) => {
```